src/optimize/scalar/bound.rs

```diff
@@ -0,0 +1,86 @@
+use std::fmt::Debug;
+use num_traits::{Float, FromPrimitive, Pow};
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum BoundOptimizerError {
+	#[error("Invalid ratio, the ratio must be greater than 1 and smaller than 2")]
+	InvalidRatio,
+	#[error("Invalid tolerance must be greater than 1")]
+	InvalidTolerance
+}
+
+#[allow(dead_code)]
+/// An iterative bound based approach for minimization of a function.
+///
+/// # Arguments
+/// * func: Function to minimize
+/// * xi: The two values characterizing the bound
+/// * ratio: The ratio around a bound to split around
+/// * tolerance: The tolerance requirement to determine convergence
+/// * max_iter: The maximum number of iterations to loop over.
+///
+/// # Example
+/// ```
+/// use mathslib::optimize::scalar::bound_optimizers::bound_minimize;
+/// use mathslib::generals::Decimal;
+///
+/// fn case_1(x: f64) -> f64{x*x + 6.0*x + 3.0	}
+///
+/// fn main() {
+/// 	assert_eq!(bound_minimize::<f64>(case_1, -9.0, 1.0, 1.5, 1e-4, 2000).unwrap().round_dp(4), -3.0000)
+///}
+/// ```
+pub fn bound_minimize<T: Float + PartialOrd + Debug + FromPrimitive>(func: fn(T) -> T, mut x1: T, mut x2: T, ratio: T, tolerance: T, max_iter: u32) -> Result<T, BoundOptimizerError>{
```

Comment:
this funciton takes a function pointer as an argument. You could explain the expected signature of this function pointer to make it easier for the user to write their own function and use it

Reply:
Fixed in 2675f34e28a57ea3126dd08fe5d8c9f5abf0f3cc.